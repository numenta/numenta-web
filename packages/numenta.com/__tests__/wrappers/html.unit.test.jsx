import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '__tests__/__mocks__/reactContextMock'
import post from '__tests__/__mocks__/postMock'

import Html from '../../wrappers/html'

const route = {page: post}


describe('HTML Wrapper React component', () => {

  it('Renders correctly', () => {
    const HtmlStubbed = stubContext(Html, context)
    const component = renderer.create(
      <HtmlStubbed route={route}>
        <div>Hello</div>
      </HtmlStubbed>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
