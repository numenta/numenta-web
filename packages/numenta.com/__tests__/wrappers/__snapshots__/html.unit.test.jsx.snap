exports[`HTML Wrapper React component Renders correctly 1`] = `
<div
  className="html">
  <h1
    dangerouslySetInnerHTML={
      Object {
        "__html": "Calculus of the Nervous System",
      }
    } />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    } />
</div>
`;
